rust-mode-treesitter.el

```diff
@@ -5,6 +5,9 @@
 
 ;;; Code:
 
+(defvar rust-before-save-hook)
```

Comment:
Shouldn't this come from `rust-common.el` ?

Reply:
Yes, but it seems you forgot to explicitly require that. It appears that, for some reason, which escapes me now, I assumed there was a reason why this library doesn't require any other libraries.

Reply:
But `rust-mode.el` does require it: https://github.com/rust-lang/rust-mode/blob/bd7d8ed2ba324dce7e531134d405eb4a8c14d876/rust-mode.el#L17

Reply:
But `rust-mode-treesitter.el` does not require `rust-mode` or any other library.

Reply:
That's because `rust-mode.el` requires `rust-mode-treesitter.el`: https://github.com/rust-lang/rust-mode/blob/bd7d8ed2ba324dce7e531134d405eb4a8c14d876/rust-mode.el#L74

Reply:
And *that* is why we need to declare this variable. When we compile `rust-mode-treesitter.el`, the compiler doesn't know that at runtime, `rust-mode-treesitter` is loaded because it is required in `rust-mode`, and that it is assumed the user (or some third-party code) does *not* load/require `rust-mode-treesitter` directly. So we have to tell the compiler about it, so it can stop being alarmist about it, and so that we don't get used to it outputting warnings and thus end up ignoring them.

However, I don't think that is right way to do it. This mode *does* depend other parts of this package, so it should depend on them using `require`. This is complicated by the fact that you also want to load this file, whenever `rust-mode` is loaded (modulo some conditions, that are not relevant to this discussion).

You *can* have it both ways, by moving `(require 'rust-mode-treesitter)` below `(provide 'rust-mode)` (as is already being done for `(require 'rust-utils)`).

Reply:
`rust-prog-mode` should also require `rust-mode`.

Reply:
> You can have it both ways, by moving (require 'rust-mode-treesitter) below (provide 'rust-mode) (as is already being done for (require 'rust-utils)).

Can you update your PR like that ? This whole require thing has caused many issues that it makes me nervous in general. Eg: https://github.com/rust-lang/rust-mode/issues/520, https://github.com/rust-lang/rust-mode/issues/527

Reply:
I've pushed an update.

Reply:
Thanks, I did a quick testing and this it looks good. I will use it throughout the day and see if any issue pops up.